### ISSUE TITLE
Multiple Functions projects on the same Workspace

### ISSUE THREAD

**Issue by galvesribeiro** [Microsoft/vscode-azurefunctions#1081]
Hello!

I got that the extension now uses VSCode new workspace feature rather than plain old .sln files. 

The problem is, VSCode settings are now embedded on a per project basis on its own `.vscode` directory making things harder to maintain.

I'm trying to have the following workspace to work:

![image](https://user-images.githubusercontent.com/4714040/54166116-684e4100-4442-11e9-873a-825fc44f499c.png)

The idea is to have multiple Functions apps under the same workspace using a single `.code-workspace` file and start/build everything when I press F5 (yeah, I've changed the Host port on each one and also added the `compounds` section as the picture show).

The problem is that if I do that, build fail miserably since it tried to build the same project multiple times and the files are in use. I guess MSBuild is not seeing each one of those projects as different ones.

I'm trying to make the same behaviour for both the developers who have VS2017 on Windows using the `.sln` as the ones on Mac OSX using VSCode and its workspace files.

Can someone shed a light on how can I achieve that?

Thanks!

**Comment by EricJizbaMSFT** [comment]
Interesting... I looked into this and couldn't get it working with the existing extension. We'll have to make changes on our side

**Comment by galvesribeiro** [comment]
Yeah @EricJizbaMSFT 

This is what we would expect from the extensions that adopt multi-rooted workspaces.

For now we would keep the project directory structure as before, having a `.sln`, and all projects under `src/` and `test/` so we keep VS2017 working and then open every function project on its own VSCode window and build/run that individually along with any solution-wide project dependencies.

Looking forward to have the fix at least on Insider track.

Thanks for the reply!

**Comment by EricJizbaMSFT** [comment]
@galvesribeiro I submitted a PR to fix this scenario. Here's our instructions for installing an insider extension: https://aka.ms/azCodeInsiders. I'll let you know once it's in master, or you can try a build from my branch: https://dev.azure.com/ms-azuretools/AzCode/_build/results?buildId=4041

I created a [sample repo](https://github.com/EricJizbaMSFT/MultiFuncApp) that was working for me. [This commit](https://github.com/EricJizbaMSFT/MultiFuncApp/commit/df9dfa8b1ccbf1d6577e4963835481aa36043da2) in particular will probably be most helpful for configuring your project.

**Comment by galvesribeiro** [comment]
Awesome @EricJizbaMSFT! 

I'll try it first thing in the morning and let you know. 

Thank you very much for the help! :)

**Comment by galvesribeiro** [comment]
Hello @EricJizbaMSFT! Sorry for the delay.

I can confirm that things are working perfectly with the insider extension!

Just curious... Will the extension update itself when you publish the release package? Or will I have to uninstall it?

Thank you! I really appreciate the help!

**Comment by galvesribeiro** [comment]
I've found an issue @EricJizbaMSFT 

If your project starts to grow, looks like the C# extension don't work pretty well. I have to keep switching the "active" project on the status bar in order to get intellisense/typeinfo/codelenses etc... 

It is like if VSCode don't detect the project as a C# project.

Idk if this is a VSCode workspace issue, an Omnisharp issue, or just another issue from the insider extension...

When another project is activated (look on the status bar GP.EventSourcing):

![image](https://user-images.githubusercontent.com/4714040/54482047-28110900-481c-11e9-9c59-519bda57c7d0.png)

When the project activated is the one the file is in (GP.EventStore):

![image](https://user-images.githubusercontent.com/4714040/54482073-74f4df80-481c-11e9-8f6d-688b1c9637a6.png)

**Comment by EricJizbaMSFT** [comment]
Yeah I'm not sure there's anything I can do about that. It feels like a problem with the C# extension (or even VS Code) itself

**Comment by galvesribeiro** [comment]
@EricJizbaMSFT I think we are with similar issue now with TypeScript functions when they are in the same workspace as the C# ones. Did you tried that scenario?

**Comment by galvesribeiro** [comment]
Nvm. After some changes on the files similar to what you did on the C# ones I was able to make it work. 

I think it should be documented somewhere that we require those changes.

Thanks!

**Comment by EricJizbaMSFT** [comment]
@galvesribeiro we just released an update to the extension including these fixes. Your extension should update automatically, but just check that you get v0.16.0 instead of v0.15.1-alpha.

I tweaked the above instructions a little bit when I wrote the "official" docs, but my recommendations from above should continue to work. See the wiki here: https://github.com/Microsoft/vscode-azurefunctions/wiki/Multiple-function-projects
